src/applications/caregivers/components/FormFields/FacilitySearch.jsx

```diff
@@ -15,11 +15,23 @@ const FacilitySearch = props => {
   const [error, setError] = useState(null);
   const [facilities, setFacilities] = useState([]);
 
-  const listProps = useMemo(() => ({ ...props, facilities, query }), [
-    facilities,
-    props,
-    query,
-  ]);
+  const facilityListProps = useMemo(
+    () => {
+      const { onChange, ...restOfProps } = props;
+      const setSelectedFacilities = facilityId => {
+        const facility = facilities.find(f => f.id === facilityId);
+        onChange({ veteranSelected: facility });
+      };
+      return {
+        ...restOfProps,
+        value: restOfProps?.formData?.veteranSelected?.id,
```

Comment:
Had to explicitly pass the facility id as the value so the FacilityList loads the selected value.